### PR TITLE
feat: use metadata protocol for awaiting connection to remote peer

### PR DESCRIPTION
**Pull request by danisharora099**
## Problem

Parent/origin: #https://github.com/waku-org/js-waku/pull/1756

There is a requirement to wait for ~50/100ms after a `peer:connect` event is fired as the Metadata Protocol is an async operation for which we need to wait to populate shard info, or we are disconnected if there is a cluster ID mismatch.

We need to wait for the Metadata protocol to complete before we can assume the peer is ready to be used to open connections reliably.

## Solution

- Add check for successful metadata handshakes in `waitForRemotePeer`
- Also make fixes around enabling sharding for nwaku nodes (https://github.com/waku-org/nwaku/issues/2259)

## Notes

Can still foresee a case where a peer that is connected is used for a protocol instantly, thus creating a race condition. 
Perhaps a more protocol-knit check should be added. Will also be helped with #1463

### CODE REVIEW

**Review by adklempner — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "cSpell.enabled": true,
   "editor.defaultFormatter": "dbaeumer.vscode-eslint",
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (fryorcraken) [inline .vscode/settings.json:6]: would be better to do a separate pr for lint/format changes

**Review by fryorcraken — COMMENTED** [review]
I would suggest to have an event for the metadata protocol such as "shard info received" and use this as a way to track whether the protocol has been run.

**Comment by weboko** [comment]
I think this PR should be prefixed with `feat!:`

File: .vscode/settings.json (hunk 1 above)
Comment (danisharora099) [reply .vscode/settings.json]: accidental change slipped in. reverted: 3475e167aacb79718bf4b992933cb0748739b679

**Comment by danisharora099** [comment]
> I think this PR should be prefixed with `feat!:`

changed to `feat` -- how do you think this is a breaking change that would require the `!`?

**Comment by danisharora099** [comment]
> I would suggest to have an event for the metadata protocol such as "shard info received" and use this as a way to track whether the protocol has been run.

This way of checking was used because on the off chance the service node does not send a request to us, or takes more time, we will initiate a request and ensure/enforce that we are on the same shard. 
Listening on an event listener can take more time since we are expecting the node to reach out to us.

Let me know if you think otherwise @fryorcraken

**Comment by danisharora099** [comment]
> > I would suggest to have an event for the metadata protocol such as "shard info received" and use this as a way to track whether the protocol has been run.
> 
> This way of checking was used because on the off chance the service node does not send a request to us, or takes more time (which it is gauged by manual tests), we will initiate a request and ensure/enforce that we are on the same shard. Listening on an event listener can take more time since we are expecting the node to reach out to us.
> 
> Let me know if you think otherwise @fryorcraken

Merging this PR for now. Happy to address the proposed change of moving to an event-based functionality as a followup.
